src/Features/CSharp/Portable/SignatureHelp/InitializerExpressionSignatureHelpProvider.cs

```diff
@@ -0,0 +1,130 @@
+               token.ValueText.Length == 1 &&
+               IsTriggerCharacter(token.ValueText[0]) &&
+               token.Parent is InitializerExpressionSyntax;
+
+        private static bool IsInitializerExpressionToken(InitializerExpressionSyntax expression, SyntaxToken token)
+            => expression.Span.Contains(token.SpanStart) && token != expression.CloseBraceToken;
+
+        protected override async Task<SignatureHelpItems> GetItemsWorkerAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (!TryGetInitializerExpression(root, position, document.GetLanguageService<ISyntaxFactsService>(), triggerInfo.TriggerReason, cancellationToken, out var initializerExpression))
+            {
+                return null;
+            }
+
+            var semanticModel = await document.GetSemanticModelForNodeAsync(initializerExpression, cancellationToken).ConfigureAwait(false);
+            var within = semanticModel.GetEnclosingNamedTypeOrAssembly(position, cancellationToken);
+            if (within == null)
+            {
+                return null;
+            }
+
+            // get the regular signature help items
+            var symbolDisplayService = document.GetLanguageService<ISymbolDisplayService>();
+            var parentOperation = semanticModel.GetOperation(initializerExpression.Parent, cancellationToken) as IObjectOrCollectionInitializerOperation;
+            if (parentOperation == null)
+            {
+                return null;
+            }
+
+            var parentType = parentOperation.Type;
+            if (parentType == null)
+            {
+                return null;
+            }
+
+            var addSymbols = semanticModel.LookupSymbols(
+                position, parentType, WellKnownMemberNames.CollectionInitializerAddMethodName, includeReducedExtensionMethods: true);
+
+            // We want all the accessible '.Add' methods that take at least two arguments. For
+            // example, say there is:
+            //
+            //      new JObject { { $$ } }
+            //
+            // Technically, the user could be calling the `.Add(object)` overload in this case.
+            // However, normally in that case, they would just supply the value directly like so:
+            //
+            //      new JObject { new JProperty(...), new JProperty(...) }
+            //
+            // So, it's a strong signal when they're inside another `{ $$ }` that they want to
+            // call the .Add methods that take multiple args, like so:
+            //
+            //      new JObject { { propName, propValue }, { propName, propValue } }
+
+            var addMethods = addSymbols.OfType<IMethodSymbol>()
+                                       .Where(m => m.Parameters.Length >= 2)
```

Comment:
😟 I'm not sure we should be excluding valid calls here. @dotnet/roslyn-ide for input on this.

Reply:
I'm amenable to undoing this.  But, frankly, i really don't think it's idiomatic or helpful to include this information.

Reply:
Also, this is not really core to the design here.  My preference would be: let's try this out.  if it turns out this a problem in the future for somebody, we can relax the check.

Reply:
So, what is the benefit of excluding `Parameter.Length == 1`?

Reply:
Yeah, I'm not sure I get this either. I think I've seen places where people might have both a one-argument and two-argument Add, where the latter is an extension method that does something else. In that case I could totally imagine somebody doing `{ { singleArgCall}. { twoArgCall1, twoArgCall2 } }` and mixing back and forth like that.

Reply:
it avoids having sig help show you the *less* likely item.  i.e. take the JObject case.  the majority of times you care about the multi-arg option.  But you'll instead see the single-arg option.  I thought that was clear in the comment i have in the code :)

Reply:
I'll provide an image to help clear this up:

![image](https://user-images.githubusercontent.com/4564579/65647256-ffc84b00-dfb2-11e9-89cc-f21f8d05df2b.png)

This is actually unhelpful.  As someone invoking the collection initializer syntax (i.e. `new JObject { {` and not `new JObject {`, i want the multi-param `Add` method sig help.  I have to explicitly get to that here since i have the not-wanted AddMethod overload.

My general argument is this, practically 100% of the time users will either want to do this:

`new JObject { a, b, c, d }`

or this:

`new JObject { { a, b },  { c, d } }`

or more rarely

`new JObject { a, { b, c }, d, { e, f } }`

I don't think users particularly want to do this:

`new JObject { { a }, { b }, { c, d } }`

Indeed, as shown above, experts in the language weren't even aware this was allowed.

This is the IDE, not the compiler.  It's not about what's legal.  It's about what's the most useful to users.  To me, targeting the vastly more likely cases, at the expense of the unlikely ones is 100% ok with me.

And, as a reminder, this is sig-help, not completion.  So it's not like leaving something out breaks typing. 

Finally, this is a new feature being added.  It's not like users critically depended on this info (since we've had this lang feature for 10+ years and never had sig help).  Users aren't going to be particularly hurt by supporting the most common case and eschewing the less common one.

Reply:
> In that case I could totally imagine somebody doing `{ { singleArgCall }, { twoArgCall1, twoArgCall2 } }` and mixing back and forth like that.

Yes, i can totally imagine it to.  But i also think it's literally the 0.01% case :)  And it feels super wrong to me to hurt the 99.99% case for that one.

Reply:
Thank you for explanations, @CyrusNajmabadi ! I'm OK with the proposal. Let us apply this `m.Parameters.Length >= 2` condition, and see if we get any concerns by using it.